src/Altinn.App.Analyzers/MetadataAnalyzer.cs

```diff
@@ -0,0 +1,91 @@
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Altinn.App.Analyzers;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+internal sealed class MetadataAnalyzer : DiagnosticAnalyzer
+{
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => Diagnostics.All;
+
+    public override void Initialize(AnalysisContext context)
+    {
+        // var configFlags = GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics;
+        var configFlags = GeneratedCodeAnalysisFlags.None;
+        context.ConfigureGeneratedCodeAnalysis(configFlags);
+        context.EnableConcurrentExecution();
+
+        context.RegisterCompilationAction(OnCompilation);
+        // System.Diagnostics.Debugger.Launch();
+    }
+
+    private void OnCompilation(CompilationAnalysisContext context)
+    {
+        if (
+            !context.Options.AnalyzerConfigOptionsProvider.GlobalOptions.TryGetValue(
+                "build_property.projectdir",
+                out var projectDir
+            )
+        )
+        {
+            context.ReportDiagnostic(Diagnostic.Create(Diagnostics.ProjectNotFound, Location.None));
+            return;
+        }
+
+#pragma warning disable RS1035 // Do not use APIs banned for analyzers
+        // On compilation end, it is not totally unreasonable to read a file
```

Comment:
Working around the issue of all apps adding `<AdditionalFiles Include="app/config/applicationmetadata.json" />` (or a wildcard for all json files if it works.)

See: https://github.com/dotnet/roslyn/blob/main/docs/analyzers/Using%20Additional%20Files.md

Reply:
Yep! We could make a TODO for v9 to add that into the project files through the CLI. I don't think we can do anything about that now?